### PR TITLE
enable file and object store smoke test and fix naming conventions

### PR DESCRIPTION
**Pull request by dangula**
Enabling file system and object store smoke test. Also fixing variable names to follow go naming convention.

### CODE REVIEW

File: e2e/framework/manager/rook_test_infra.go [hunk 1]
```diff
@@ -170,30 +170,30 @@ func (r *rookTestInfraManager) copyImageToNode(containerId string, imageName str
 		cmdOut := utils.ExecuteCommand(objects.CommandArgs{Command: "docker", CmdArgs: []string{"pull", imageName}})
 
 		if cmdOut.Err != nil || cmdOut.ExitCode != 0 {
-			return fmt.Errorf("Failed to pull image")
+			panic(fmt.Errorf("Failed to pull image"))
```
Comment (jbw976) [inline e2e/framework/manager/rook_test_infra.go]: consider including `imageName` in the panic output (this applies to other panics below too)

File: e2e/framework/utils/k8s_helper.go [hunk 2]
```diff
@@ -251,3 +251,60 @@ func (k8sh *K8sHelper) IsThirdPartyResourcePresent(tprname string) bool {
...
 }
+
+func (k8sh *K8sHelper) GetPodDetails(podNamePattern string, namespace string) (string, error) {
+	cmdArgs := []string{"get", "pods", "-l", "app=" + podNamePattern, "-o", "wide"}
+	if namespace != "" {
+		cmdArgs = append(cmdArgs, []string{"-n", namespace}...)
+	}
+	sout, serr, status := ExecuteCmd("kubectl", cmdArgs)
+	if status != 0 {
+		return serr, fmt.Errorf("Cannot find service")
```
Comment (jbw976) [inline e2e/framework/utils/k8s_helper.go]: error message refers to "service" but this function is for pod details.  Also consider including the `podNamePattern` and `namespace` in the error message.

File: e2e/framework/utils/k8s_helper.go [hunk 3]
```diff
@@ -251,3 +251,60 @@ func (k8sh *K8sHelper) IsThirdPartyResourcePresent(tprname string) bool {
...
+	}
+	return sout, nil
+}
+
+func (k8sh *K8sHelper) GetPodHostId(podNamePattern string, namespace string) (string, error) {
+	data, err1 := k8sh.GetPodDetails(podNamePattern, namespace)
```
Comment (jbw976) [inline e2e/framework/utils/k8s_helper.go]: why `err1` instead of just `err`?

File: e2e/framework/utils/k8s_helper.go [hunk 4]
```diff
@@ -251,3 +251,60 @@ func (k8sh *K8sHelper) IsThirdPartyResourcePresent(tprname string) bool {
...
+	}
+
+	// Handle case when no data is returned
+	lines := strings.Split(data, "\n")
+	if len(lines) <= 1 {
+		return "NOT FOUND", fmt.Errorf("Cannot find Pod with Name matching %s", podNamePattern)
+	}
+
+	// Handle case when no data is returned
+	lines = lines[1 : len(lines)-1]
```
Comment (jbw976) [inline e2e/framework/utils/k8s_helper.go]: looks like we're ignoring the 1st line, probably because that's the column headers? another way to do this would be to specify `--no-headers=true`.

File: e2e/framework/manager/rook_test_infra.go [hunk 5]
```diff
@@ -34,7 +34,7 @@ var (
 )
 
 const (
-	temp_image_file_name           = "temp_image.tar"
+	tempImageFileName              = "temp_image.tar"
```
Comment (travisn) [inline e2e/framework/manager/rook_test_infra.go]: in general we would want to limit our changes in a PR to the topic of the PR. For example, the cleanup changes would go in their own PR rather than being included in this one.

File: e2e/framework/utils/k8s_helper.go [hunk 6]
```diff
@@ -251,3 +251,60 @@ func (k8sh *K8sHelper) IsThirdPartyResourcePresent(tprname string) bool {
...
+	lineRawdata := strings.Split(lines[0], "  ")
+	var r []string
+	for _, str := range lineRawdata {
+		if str != "" {
+			r = append(r, strings.TrimSpace(str))
+		}
+	}
+
+	//get host Ip of the pod
+	cmdArgs := []string{"get", "pods", r[0], "-o", "json"}
```
Comment (jbw976) [inline e2e/framework/utils/k8s_helper.go]: this may be easier using output type `jsonpath` instead.  Essentially you pass a query path to use on the returned JSON for the field you care about, and then you could skip the deserialization and querying you are doing below.

File: e2e/framework/utils/s3_operations.go [hunk 7]
```diff
@@ -2,29 +2,47 @@ package utils
...
 
-	svc := s3.New(session.New(), cfg)
+	awsConfig := aws.NewConfig().
+		WithRegion("other-v2-signature").
+		WithCredentials(creds).
+		WithEndpoint(endpoint).
+		WithS3ForcePathStyle(true).
+		WithDisableSSL(true)
+
+	ses := session.New()
```
Comment (jbw976) [inline e2e/framework/utils/s3_operations.go]: some comments about what is being done here may be useful.

File: e2e/tests/smoke/object_storage_smoke_test.go [hunk 8]
```diff
@@ -141,30 +132,27 @@ func (suite *ObjectStorageTestSuite) TestObjectStorage_SmokeTest() {
...
 
 }
 
-func objetTestcleanup() {
-	sc, _ := CreateSmokeTestClient(enums.Kubernetes)
-	conninfo, _ := sc.GetObjectStoreConnection(userid)
-	port, _ := sc.GetRGWPort()
-	//TODO - need to figure out how to expose rgw endpoint and get host ip.
-	s3endpoint := "http://172.17.4.101:" + port
+func objetTestcleanup(h *SmokeTestHelper) {
```
Comment (jbw976) [inline e2e/tests/smoke/object_storage_smoke_test.go]: typo: `objet...`

File: e2e/tests/smoke/smoke_test_helper.go [hunk 9]
```diff
@@ -408,14 +408,14 @@ func (h *SmokeTestHelper) DeleteObjectStoreUser() (string, error) {
...
 	switch h.platform {
 	case enums.Kubernetes:
-		rawdata, err := h.k8sHelp.GetService("rgw-external")
+		hostip, err := h.k8sHelp.GetPodHostId("rgw", "rook")
 		if err != nil {
-			return "port not found", err
+			return "RGW pods not found/object store possibly not started ", err
 		}
-		return h.rookHelp.GetRgwServiceNodePort(rawdata)
+		return hostip + ":30001", err
```
Comment (jbw976) [inline e2e/tests/smoke/smoke_test_helper.go]: should this `30001` value be a const?

File: e2e/framework/utils/s3_operations.go [hunk 10]
```diff
@@ -22,19 +22,26 @@ type S3Helper struct {
...
 		WithS3ForcePathStyle(true).
 		WithDisableSSL(true)
 
+	//create new session
 	ses := session.New()
+
+	//create new s3 client connection
 	c := s3.New(ses, awsConfig)
+
+	// user a custom signer - need to convert newer amz s3 requests to v2 requests since
```
Comment (jbw976) [inline e2e/framework/utils/s3_operations.go]: is this comment meant to be continued? it looks like its cut off...

I had made a comment earlier on this PR but i don't see it anymore, so I'm not sure where it went...but is this complicated/additional logic in the `sign` function due to a limitation in RGW or in our s3 library?  Is there any way to avoid having to write and maintain all this logic?

Comment (dangula) [inline e2e/framework/utils/s3_operations.go]: It is due to a compatibility issues with the new s3 libraries. , whenever you make a request via aws-sdk-go, it builds a s3 request in this pattern http://{bucketName}.{Endpoint}/.... . When using non amazon object store like ceph,  DNS cannot resolve above url, (it works if your are using actual amazon url  and valid region code).  The sign function is a hack to convert s3 request to older v2 version e.g. http://endpoint/{bucketName}/...
 I found this hack on line and slimed it down to our use.  This should be a one time thing.  Our other option is to ditch was-sdk s3 libraries and find something else. Again using a our own http client is not trivial because we need to figure out all the 'amz' specific headers we need to use.,which are  unfortunately not documented or they are hard to find.

Comment (dangula) [inline e2e/framework/utils/s3_operations.go]: Also clean up the comments.

Comment (jbw976) [inline e2e/framework/utils/s3_operations.go]: Could you capture this background information better in the comments?  That's really useful to know *why* we're doing all this work in the `sign` function.

Comment (dangula) [inline e2e/framework/utils/s3_operations.go]: Actually, I am removing the stupid hack, I was doing some testing and looks like using the right region is doing the trick, Lack of documentation on s3 lead me down the wierd path earlier, I don't think we should see DNSError issues anymore.

File: e2e/framework/utils/s3_operations.go [hunk 11]
```diff
@@ -2,29 +2,38 @@ package utils
...
 	creds := credentials.NewStaticCredentials(keyId, keySecret, "")
-	_, err := creds.Get()
-	if err != nil {
-		fmt.Printf("bad credentials: %s", err)
-	}
-	cfg := aws.NewConfig().WithEndpoint(endpoint).WithRegion("us-west-1").WithCredentials(creds).WithDisableSSL(true)
 
-	svc := s3.New(session.New(), cfg)
+	// create aws s3 config, must use 'other-v2-signature' region for ceph object store
+	awsConfig := aws.NewConfig().
```
Comment (jbw976) [inline e2e/framework/utils/s3_operations.go]: this is much better without all the signing workarounds

Comment (dangula) [inline e2e/framework/utils/s3_operations.go]: alright squashed my commits and rebased with upstream/master. I ran the object store test multiple times locally and on Jenkins, the trick was to using the right aws region and use WithS3ForcePathStyle(true) option.  unfortunately a lot of s3 stuff was trial and error due to poor documentation
